cryptographic_estimators/SDEstimator/SDAlgorithms/bjmm_plus.py

```diff
@@ -0,0 +1,261 @@
+# ****************************************************************************
+# Copyright 2023 Technology Innovation Institute
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# ****************************************************************************
+
+
+from ...base_algorithm import optimal_parameter
+from ...SDEstimator.sd_algorithm import SDAlgorithm
+from ...SDEstimator.sd_problem import SDProblem
+from ...SDEstimator.sd_helper import _gaussian_elimination_complexity, _mem_matrix, _list_merge_complexity, min_max, \
+    binom, log2, ceil, inf, _list_merge_async_complexity
+from types import SimpleNamespace
+from ..sd_constants import *
+
+
+class BJMM_plus(SDAlgorithm):
+    def __init__(self, problem: SDProblem, **kwargs):
+        """
+        Complexity estimate of BJMM+ algorithm in depth 2
+
+        [MMT11] May, A., Meurer, A., Thomae, E.: Decoding random linear codes in  2^(0.054n). In: International Conference
+        on the Theory and Application of Cryptology and Information Security. pp. 107–124. Springer (2011)
+
+        [BJMM12] Becker, A., Joux, A., May, A., Meurer, A.: Decoding random binary linear codes in 2^(n/20): How 1+ 1= 0
+        improves information set decoding. In: Annual international conference on the theory and applications of
+        cryptographic techniques. pp. 520–536. Springer (2012)
+
+        Additionally, the improvements by [EZ23]_ are incorporated.
```

Comment:
should be add the top as this is the reason why it is called "plus". Also the MMT11 and BJMM12 should be replaced by the references [MMT11]_ and [BJMM12]_ (also in the other BJMM implementation)